spam/spam_tutorial.py

```diff
@@ -140,234 +145,406 @@
 # * *Third-party models*: using an pre-trained model (usually a model for a different task than the one at hand)
 # * *Distant supervision*: using external knowledge base
 # * *Crowdworker labels*: treating each crowdworker as a black-box function that assigns labels to subsets of the data
+
+# %% [markdown]
+# ### Recommended practice for LF development
+
+# %% [markdown]
+# Typical LF development cycles include multiple iterations of ideation, refining, evaluation, and debugging.
+# A typical cycle consists of the following steps:
+#
+# 1. Look at examples to generate ideas for LFs
+# 1. Write an initial version of an LF
+# 1. Estimate its performance by looking at examples in the training set or development set
+# 1. Refine and debug to improve coverage or accuracy as necessary
 #
-# The process of **developing LFs** is iterative and usually consists of:
-# * Writing a function
-# * Estimating its performance by looking at labeled examples in the training set or dev set
-# * Iterating to improve coverage or accuracy as necessary.
+# Labeling function development is a creative process.
+# However, there are a few principles that we highly recommend using when developing labeling functions.
 #
-# Balancing accuracy and coverage for specific labeling functions as well as the overall set of LFs developed is often a trade-off, and depending on the use case, users may tend to prefer one over the other.
+# * **Use the unlabeled training set for ideation whenever possible.** Reserve labeled examples for evaluation in order to avoid overfitting LFs or models.
```

Comment:
I don't like this point for several reasons:

- "ideation"... what were you on the YDN or something?
- More importantly: this could be construed as implying that not looking at the test set is a 'recommended best practice', vs. something that you absolutely do not do
- Also even the intended point- that you should look at the training set not the dev set- is not something I consider unambiguous.  I would say do use the dev set for ideation, that's what it's for / where it could be created

Overall I'd drop this point